### DISCUSSION TITLE
Dynamic Time Warping

### DISCUSSION THREAD

**Discussion by wangkev** [TDAmeritrade/stumpy#284] [Q&A]
Thank you for the awesome library!

Wanted to follow up the discussion of usage of matrix profiles with dynamic time warping for motif discovery. Seems like the there was recently a [paper](https://www.cs.ucr.edu/~eamonn/DTWMotifsSWAMP_Expanded.pdf) published on the subject. Is this on the roadmap to support?

Past Discussions:
- https://github.com/TDAmeritrade/stumpy/issues/233

**Comment by seanlaw** [comment]
Thank you for your question @wangkev. I am aware of this paper but this it not currently on our roadmap. Unfortunately, I have limited experience with DTW and my feeling is that, without assistance, implementing DTW is beyond the scope of this package. It seems that there are a lot of ways to optimize DTW and my fear is that we can spend all of our bandwidth on chasing a moving target. 

Of course, this is my own limitation and we would welcome others to contribute to this package.

**Comment by seanlaw** [comment]
In fact, I think that this would be more appropriate as a separate package that might depend on STUMPY. While the capability might be useful/valuable to users, it is also important (from a long term maintainability standpoint) that STUMPY not try to be everything to everyone.

**Comment by seanlaw** [comment]
@wangkev If it's okay with you, I'd like to close this for now. Feel free to re-open, comment, or open up a new issue if you have any further questions.

**Comment by wangkev** [comment]
@seanlaw Understood, thanks!

**Reply by Djangu-algo** [reply]
This may or may not be helpful but at least interesting on the topic of this discussion and possible future implementation ["Everything you know about Dynamic Time Warping is Wrong"](https://www.cs.ucr.edu/~eamonn/DTW_myths.pdf)

**Comment by jonahpearl** [comment]
Also interested in this...@wangkev did you ever find or make anything in python for implementing the algs in the paper?

**Comment by saeed349** [comment]
Would also love to see this implemented in stumpy
